### ISSUE TITLE
Custom color palete not showing with wp.data.select('core/editor').getEditorSettings().colors

### ISSUE THREAD

**Issue by huubl** [WordPress/gutenberg#26504]
Hi,

When I use the console and do `wp.data.select('core/editor').getEditorSettings().colors` it's showing the default colors and not the ones I added by  `add_theme_support('editor-color-palette', [...])`.

Why, and how to get the custom palette colors?

**Comment by talldan** [comment]
@smeedijzer-internet Does `wp.data.select('core/block-editor').getSettings().colors` return the correct colors?

**Comment by smeedijzer-internet** [comment]
No, it only shows the default color scheme.

**Comment by CrackAjax** [comment]
I am also lacking the custom theme colors in the settings object. Here are the properties I see:
<img width="1085" alt="Screen Shot 2020-11-06 at 11 05 25 AM" src="https://user-images.githubusercontent.com/8129248/98399724-16e3c400-2020-11eb-9d57-88204d1b8dc0.png">

**Comment by mkkeck** [comment]
How can this be fixed?

**Comment by andrewstaffell** [comment]
Just came up against this problem which stops `theme.json` colours from being applied in custom blocks. Any progress?

**Comment by skorasaurus** [comment]
May want to consider this closed since the new recommended way of adding a custom color palette is now through theme.json (ref https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#css-custom-properties-presets-custom) instead of through add_theme_support

**Comment by hadamlenz** [comment]
i love searching "wp data select get colors" and this coming up first with zero explanation and its closed

**Comment by skorasaurus** [comment]
@hadamlenz 

I understand your frustration; re-reading over my comment, I don't understand why I even closed it based on that explanation, I'm sorry about that.

**Comment by skorasaurus** [comment]
Although for me when I issue wp.data.select('core/editor').getEditorSettings().colors in the console I am able to obtain the colors (And only my custom color pallete) 

Could someone see if they can produce this ? 

![image](https://github.com/user-attachments/assets/51b8b6f2-6cde-414a-af7b-0a698e5e4411)

**Comment by hadamlenz** [comment]
I am seeing the same, I was able to get the theme colors with `useSettings("color.palette.theme");`  and then mash the results together but I'm still looking for the builtin preset colors.  Is this the opposite of the original problem?

**Comment by Adi-ty** [comment]
I've tested this across different theme configurations and can confirm the behavior has evolved since the original issue was reported:

The `wp.data.select('core/editor').getEditorSettings().colors` selector now **correctly returns custom color palettes** in all scenarios:

I added this `add_theme_support` configuration with 25 colors to my Twenty Twenty-Five theme:

```php
add_theme_support(
    'editor-color-palette',
    array(
        array(
            'name'  => __( 'Dark Gray', 'twentyfifteen' ),
            'slug'  => 'dark-gray',
            'color' => '#111',
        ),
        array(
            'name'  => __( 'Light Gray', 'twentyfifteen' ),
            'slug'  => 'light-gray',
            'color' => '#f1f1f1',
        ),
        // ... [23 more colors] ...
    )
);
```

### **Results:**

#### **Block Themes with theme.json colors:**
- Returns colors defined in `theme.json`

<img width="664" alt="Image" src="https://github.com/user-attachments/assets/b7a0cf39-df72-4a7e-ba7f-05313efe5ccd" />

#### **Block Themes with NO colors in theme.json:**  
- Falls back to `add_theme_support('editor-color-palette')` colors

<img width="877" alt="Image" src="https://github.com/user-attachments/assets/50b5957b-9579-42b7-b789-ff596dd003bc" />


#### **Priority System Works Correctly:**
When both `theme.json` and `add_theme_support` colors are present:
- `theme.json` colors take precedence 
- If `theme.json` has no colors, falls back to `add_theme_support` 
- Fallback to WordPress defaults when no custom colors defined

**Comment by SirLouen** [comment]
> #### **Priority System Works Correctly:**
> When both `theme.json` and `add_theme_support` colors are present:
> 
> * `theme.json` colors take precedence
> * If `theme.json` has no colors, falls back to `add_theme_support`
> * Fallback to WordPress defaults when no custom colors defined

Yes, this is working exactly like this, but feels counterintuitive to me and I think it's the reason of why this report was opened originally.
Shouldn't custom colours get priority over `theme.json` ones?  (or minimally pushed into the default array)

@t-hamano what do you think?

**Comment by t-hamano** [comment]
> #### **Priority System Works Correctly:**
> When both `theme.json` and `add_theme_support` colors are present:
> - `theme.json` colors take precedence 
> - If `theme.json` has no colors, falls back to `add_theme_support` 
> - Fallback to WordPress defaults when no custom colors defined 

All of this is correct and documented behavior:

https://developer.wordpress.org/block-editor/how-to-guides/themes/global-settings-and-styles/#backward-compatibility-with-add_theme_support

> If the `theme.json` contains any settings, these will take precedence over the values declared via `add_theme_support`. This is the complete list of equivalences:

I couldn't find any discussion about the decision to use this specification, but it seems that this has been the case for at least five years since theme.json was experimental.

https://github.com/WordPress/gutenberg/pull/22518/files?diff=unified&w=0#diff-c500b97cdaef1b3b38a3d55a0186fed1ea3f797950888760b4d632d92f94faeaR148

I suggest closing this issue as resolved and discussing color priority in a separate issue if necessary.

**Comment by SirLouen** [comment]
> All of this is correct and documented behavior:

> https://developer.wordpress.org/block-editor/how-to-guides/themes/global-settings-and-styles/#backward-compatibility-with-add_theme_support

Good information.

> I suggest closing this issue as resolved and discussing color priority in a separate issue if necessary.

OK, probably it can be closed, if anyone who has participated here has any interest in planning a new priority, can open a ticket as you suggest.

Moreover, the `needs-testing` tag can be removed
